packages/xo-server/src/xapi/mixins/pool.mjs

```diff
@@ -89,46 +90,61 @@ const methods = {
 
     let hasRestartedOne = false
     // Restart all the hosts one by one
-    for (const host of hosts) {
-      const hostId = host.uuid
-      if (ignoreHost && ignoreHost(host)) {
-        continue
-      }
-
-      // This is an old metrics reference from before the pool master restart.
-      // The references don't seem to change but it's not guaranteed.
-      const metricsRef = host.metrics
-
-      await this.barrier(metricsRef)
-      await this._waitObjectState(metricsRef, metrics => metrics.live)
-
-      const getServerTime = async () => parseDateTime(await this.call('host.get_servertime', host.$ref)) * 1e3
-      log.debug(`Evacuate host ${hostId}`)
-      await this.clearHost(host)
-
-      if (beforeRebootHost) {
-        await beforeRebootHost(host)
+    await Task.run({ properties: { name: `Restart hosts`, total: hosts.length } }, async () => {
+      let done = 0
+      Task.set('progress', 0)
+      Task.set('done', 0)
```

Comment:
Why not initializing them directly in `properties`?

IMHO, it's better as it ensure the task will not exist without these properties, even for a split second.